jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4561,20 +4561,77 @@ public boolean isNestmateOf(Class<?> that) {
 /*[ENDIF] Java11 */
 
 /*[IF Java12]*/
+	/**
+	 * Create class of an array. The component type will be this Class instance.
+	 * 
+	 * @return array class where the component type is this Class instance
+	 */
 	public Class<?> arrayType() {
-		throw new UnsupportedOperationException("Stub for Java 12 compilation (Jep334)");
+		if (this == void.class) {
+			throw new IllegalArgumentException();
+		}
+		return arrayTypeHelper();
 	}
 
+	private native Class<?> arrayTypeHelper();
```

Comment:
The more typical naming convention is adding `impl` to the method name

Reply:
I just pushed the rename.